src/main/java/org/jitsi/videobridge/LipSyncHack.java

```diff
@@ -434,6 +445,107 @@ public void onRTPTranslatorWillWriteVideo(
     }
 
     /**
+     * {@inheritDoc}
+     */
+    @Override
+    public PacketTransformer getRTPTransformer()
+    {
+        return this;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public PacketTransformer getRTCPTransformer()
+    {
+        return null;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void close()
+    {
+
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public RawPacket[] reverseTransform(RawPacket[] pkts)
+    {
+        return pkts;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public RawPacket[] transform(RawPacket[] pkts)
+    {
+        if (pkts == null || pkts.length == 0)
+        {
+            return pkts;
+        }
+
+        RawPacket[] extra = null;
+
+        for (int i = 0; i < pkts.length; i++)
+        {
+            if (pkts[i] == null)
+            {
+                continue;
+            }
+
+            long ssrc = pkts[i].getSSRCAsLong();
+            if (!ssrcsWithoutBlackKeyframes.isEmpty()
+                && ssrcsWithoutBlackKeyframes.contains(ssrc))
+            {
+                ssrcsWithoutBlackKeyframes.remove(ssrc);
+
+                int seqNum = pkts[i].getSequenceNumber();
+                long ts = pkts[i].getTimestamp();
+                RawPacket[] kfs = new RawPacket[MAX_KEY_FRAMES];
+                for (int j = 0; j < kfs.length; j++)
+                {
+                    int relativeIdx = (j - 1 - kfs.length);
+                    byte[] buf = KEY_FRAME_BUFFER.clone();
```

Comment:
The byte array holds a full RTP packet and not just the VP8 payload, if that's what you're referring to.